src/plugins/discover/public/application/angular/discover.js

```diff
@@ -557,6 +557,7 @@ function discoverController($route, $scope, Promise) {
 
         $scope.fetchStatus = fetchStatuses.NO_RESULTS;
         $scope.fetchError = error;
+        error.attributes.index_pattern = $scope.indexPattern;
```

Comment:
How would a search embeddable on a dashboard behave?

Reply:
Maybe ES could specify the index ID in the response?
They should have this information.

Reply:
ES does specify the index but a single index can belong to more than one index pattern. And the index pattern has the painless script on either a runtime field or a scripted field.